### PR TITLE
lnav: add a few missing dependencies

### PR DESCRIPTION
**Pull request by symphorien**
also add a few missing optional dependencies
enable separateDebugInfo as lnav crashes not unfrequently. 
add myself as maintainer

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Atemu** [comment]
I don't know the program enough to know what these deps are useful for but it looks reasonable to me.

Could you split the logical steps you've taken into separate commits?

**Review by Atemu — APPROVED** [review]

**Comment by symphorien** [comment]
I split the commits.

About the new dependencies here is the diff of the log outpot of ./configure:
```diff
--- logold	2024-11-23 17:33:36.605402229 +0100
+++ log	2024-11-23 17:33:33.509357416 +0100
@@ -31,7 +31,7 @@
 autoreconf: Leaving directory '.'
 patching script interpreter paths in ./configure
 ./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/sh"
-configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/i89q3v4ap60ikljgsdr8xw3ky66qywjg-lnav-0.12.3
+configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3 --bindir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/bin --sbindir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/sbin --includedir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/include --oldincludedir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/include --mandir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/share/man --infodir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/share/info --docdir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/share/doc/lnav --libdir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/lib --libexecdir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/libexec --localedir=/nix/store/2qb6126273qk4g2xmsck7c5a656jcqyy-lnav-0.12.3/share/locale
 checking for a BSD-compatible install... /nix/store/k48bha2fjqzarg52picsdfwlqx75aqbb-coreutils-9.5/bin/install -c
 checking whether build environment is sane... yes
 checking for a race-free mkdir -p... /nix/store/k48bha2fjqzarg52picsdfwlqx75aqbb-coreutils-9.5/bin/mkdir -p
@@ -85,7 +85,7 @@
 checking for bash... /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash
 checking for cargo... no
 checking for bzip2... /nix/store/y59bxw99pbld247dznzv352cqdhlybqi-bzip2-1.0.8-bin/bin/bzip2
-checking for re2c... no
+checking for re2c... /nix/store/m86phdxnx92wr71cj08xybyi4axm0pdi-re2c-3.1/bin/re2c
 checking for xz... /nix/store/g6rf9zyrz8hrgz4r7sd2hab3j1afxdkr-xz-5.6.3-bin/bin/xz
 checking for tshark... no
 checking for check-jsonschema... no
@@ -118,8 +118,7 @@
 configure: Trying to link with tinfo
 checking for library containing cur_term... -ltinfo
 configure: Linking with tinfo
-checking for library containing Gpm_Open... no
-configure: WARNING: libgpm not found. If build fails later consider installing gpm dev package
+checking for library containing Gpm_Open... -lgpm
 checking for execinfo.h... yes
 checking for pty.h... yes
 checking for util.h... no
@@ -133,11 +132,9 @@
 checking for working ncursesw.h... no
 checking for working ncurses.h... yes
 checking lib archive... (testing)
-checking for archive_read_new in -larchive... no
-checking for archive.h... no
-checking for archive_read_new in -larchive... (cached) no
-checking for archive.h... no
-checking lib archive... no
+checking for archive_read_new in -larchive... yes
+checking for archive.h... yes
+checking lib archive... -larchive
 checking if PCRE2 is wanted... yes
 checking for pcre2_compile_8 in -lpcre2-8... yes
 checking for pcre2.h... yes
@@ -153,8 +150,6 @@
 checking for sqlite3_value_subtype... yes
 checking for sqlite3_error_offset... yes
 checking for sqlite3_drop_modules... yes
-configure: Checking for libgpm dependency
-checking for mousemask in -lncursesw... yes
 checking that generated files are newer than configure... done
 configure: creating ./config.status
 config.status: creating Makefile
```

File: pkgs/by-name/ln/lnav/package.nix [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+  zlib,
+  bzip2,
+  autoconf,
+  automake,
+  curl,
+  buildPackages,
+  re2c,
+  gpm,
+  libarchive,
+  nix-update-script,
```
Comment (Atemu) [inline pkgs/by-name/ln/lnav/package.nix:18]: Please separate the reformat out too.

**Review by Atemu — APPROVED** [review]

File: pkgs/by-name/ln/lnav/package.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/by-name/ln/lnav/package.nix:18]: done

**Review by Atemu — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357675`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>lnav</li>
    <li>lnav.debug</li>
  </ul>
</details>

**Comment by pcasaretto** [comment]
It seems gpm is not available on darwin, so it broke the build.
Why didn't the build system build this for all archs?

**Comment by Atemu** [comment]
In addition to ofBorg eval being slow as heck, its Darwin builders are currently also too slow to rely on.

Even if it worked, it's not immediately obvious whether a build failing is a regression or not; we have many many packages that don't eval or build on Darwin.

Please also note that Darwin is a tier 2 platform (https://github.com/NixOS/rfcs/pull/46), so support is more of a best-effort sort of thing. It's not unexpected that it breaks every now and then.

If you'd like, you can add yourself to the package's maintainers in order to be pinged to review future PRs. Most people who contribute to Nixpkgs don't have a mac, so it's good to have some variety in that regard.

The fix for this should be trivial: Make the dep optional and disabled when `stdenv.isDarwin`.

**Comment by sagikazarmark** [comment]
I opened #360350, it fixes the problem for me.

**Pull request merged**